src/hip_memory.cpp

```diff
@@ -1532,6 +1532,57 @@ __global__ void hip_copy2d_n(T* dst, const T* src, size_t width, size_t height,
 }
 }  // namespace
 
+hipError_t hipMemcpyToSymbol(const void* symbolName, const void* src,
```

Comment:
If we move the C++ implementation of these functions from hip_runtime_api.h into this .cpp file than then we won't need to duplicate the logic/code like below

Reply:
Thanks for review,
It seems hipMemcpy*Symbol* was brought to header file in order to fix dlopen issue. More details can be found in PR# https://github.com/ROCm-Developer-Tools/HIP/pull/929. I was not sure if bringing it back to .CPP may break dlopen scenario. Hence for CPP I did not change the behavior and for gcc I thought of using older implementation of these APIs.

Please let me know your thoughts